### PR TITLE
feat: feat: added vue for exports in solana

### PR DESCRIPTION
**Pull request by phoenixVS**
# Breaking Changes

N/A

# Changes

- feat:
- fix:
- chore:

# Associated Issues

closes #...

### CODE REVIEW

File: packages/solana/exports/vue.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
+import { useSnapshot } from 'valtio'
```
Comment (glitch-txs) [inline packages/solana/exports/vue.ts:1]: Valtio is a React dependent package and should not be used here. Check the `scaffold-vue` package please.

**Review by glitch-txs — CHANGES_REQUESTED** [review]

**Review by enesozturk — APPROVED** [review]

File: examples/react-solana/src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
+import { defaultWagmiConfig } from '@web3modal/solana/react/config'
+import {
+  createWeb3Modal,
+  useWeb3Modal,
+  useWeb3ModalEvents,
+  useWeb3ModalState,
+  useWeb3ModalTheme
+} from '@web3modal/solana/react'
+import { WagmiProvider } from 'wagmi'
```
Comment (enesozturk) [inline examples/react-solana/src/App.tsx:9]: Do we need wagmi for Solana example?

File: examples/react-solana/src/App.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+export default function App() {
+  // 4. Use modal hook
+  const modal = useWeb3Modal()
+  const state = useWeb3ModalState()
+  const { themeMode, themeVariables, setThemeMode } = useWeb3ModalTheme()
+  const events = useWeb3ModalEvents()
+
+  return (
+    <WagmiProvider config={wagmiConfig}>
```
Comment (enesozturk) [inline examples/react-solana/src/App.tsx:74]: Same, do we need wagmi related setup in this example?

File: examples/react-solana/package.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    "dev:example": "vite --port 3002",
+    "build:examples": "vite build"
+  },
+  "dependencies": {
+    "@web3modal/solana": "5.0.6",
+    "@tanstack/react-query": "5.24.8",
+    "react": "18.2.0",
+    "react-dom": "18.2.0",
+    "vite": "5.2.11",
+    "viem": "2.10.3",
```
Comment (enesozturk) [inline examples/react-solana/package.json:15]: Prob we don't need `viem` too.

**Review by enesozturk — APPROVED** [review]

**Review by glitch-txs — APPROVED** [review]

**Pull request merged**